File: lecture05/02_lotto/template.py
#!/usr/bin/env python3

import itertools
import sys

# Constants

LOTTO_NUMBERS = 6

# Functions

def display_combinations(s, n=LOTTO_NUMBERS):
    # TODO Version 1: Use itertools to display combinations

def combinations(s, d, k):
    # TODO Version 2: Generate combinations recursively

    # TODO Base: have a complete subset

    # TODO Recurse: with current

    # TODO Recurse: skip current

def main():
    for line, numbers in enumerate(map(str.split, sys.stdin)):
        if len(numbers) <= 1:
            break

        # TODO: Handle blank line between cases

        # TODO Version 1: Use display_combinations

        # TODO Version 2: Convert numbers to integers, sort them, and then
        # recursively generate combinations

# Main execution

if __name__ == '__main__':
    main()

File: lecture05/02_lotto/template.cpp
// lotto.cpp

#include <algorithm>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

// Functions

void combinations(vector<int> &s, vector<int> &d, size_t k) {
    // TODO: Base: have complete subset
    
    // TODO: Recurse: with current

    // TODO: Recurse: skip current
}

int read_numbers(vector<int> &v) {
    // Read line
    string line;
    if (!getline(cin, line)) {
    	return 0;
    }

    // Read k (first number)
    stringstream ss(line);
    size_t k;
    ss >> k;
    if (!k) {
    	return 0;
    }

    // Read numbers
    v.resize(k);
    for (size_t i = 0; i < k; i++) {
    	ss >> v[i];
    }

    return v.size();
}

// Main execution

int main(int argc, char *argv[]) {
    vector<int> numbers;

    for (size_t test_case = 0; read_numbers(numbers); test_case++) {
    	// TODO: Sort numbers

	// TODO: Handle blank space between test cases

	// TODO: Generate combinations recursively
    }

    return 0;
}
